tensorly/decomposition/_cp.py

```diff
@@ -214,6 +214,9 @@ def error_calc(tensor, norm_tensor, weights, factors, sparsity, mask, mttkrp=Non
                 tl.abs(norm_tensor**2 + factors_norm**2 - 2 * iprod)
             )
 
+            errchk, _, _ = error_calc(tensor, norm_tensor, weights, factors, sparsity, mask, mttkrp=None)
```

Comment:
There is a more general issue here. I added this check—whenever the mttkrp is used it checks this against the error found without using it. The result always matches with a rank 1 problem, but not with *any* rank >1 problems. So, this seems to not be restricted to complex problems—it just happens to show up in that case.